### PR TITLE
netease-cloud-music-gtk: 2.2.0 -> 2.3.0

### PR DESCRIPTION
**Pull request by Aleksanaa**
## Description of changes

~~ping @wegank for testing (thanks in advance!~~

~~The inability to play music may be due to the NetEase Cloud account being risk controlled, or it may be caused by a proxy. We need to find more people to test it.~~

Tested and works. Current version in Nixpkgs doesn't work because of https://github.com/nixos/nixpkgs/issues/271960. If you still find it unable to play music, try deleting `~/.cache/gstreamer-1.0`.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by wegank** [comment]
I don't have a NetEase Cloud account, though...

**Comment by Aleksanaa** [comment]
This is also affected by https://github.com/nixos/nixpkgs/issues/271960, and I've added a workaround.

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 287212` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 287212` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)

File: pkgs/applications/audio/netease-cloud-music-gtk/default.nix [hunk 1]
```diff
@@ -73,12 +76,17 @@ stdenv.mkDerivation rec {
...
+    gappsWrapperArgs+=(--prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath [ libsoup_3 ]}")
+  '';
+
   meta = with lib; {
     description = "A Rust + GTK based netease cloud music player";
     homepage = "https://github.com/gmg137/netease-cloud-music-gtk";
     license = licenses.gpl3Plus;
-    maintainers = with maintainers; [ diffumist ];
+    maintainers = with maintainers; [ diffumist aleksana ];
     mainProgram = "netease-cloud-music-gtk4";
```
Comment (wegank) [inline pkgs/applications/audio/netease-cloud-music-gtk/default.nix:90]:
```suggestion
    mainProgram = "netease-cloud-music-gtk4";
    platforms = platforms.linux;
```

**Comment by Aleksanaa** [comment]
The by-name check failure is irrelevant.

**Pull request merged**